backend/src/collectivites/recherches/recherches.router.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { TrpcService } from '@/backend/utils/trpc/trpc.service';
+import { PermissionService } from '@/backend/auth/authorizations/permission.service';
+import RecherchesService from '@/backend/collectivites/recherches/recherches.service';
+import { filtersRequestSchema } from '@/backend/collectivites/recherches/filters.request';
+
+@Injectable()
+export class RecherchesRouter {
+  constructor(
+    private readonly trpc: TrpcService,
+    private readonly service: RecherchesService,
+    private readonly permission: PermissionService
+  ) {}
+
+  router = this.trpc.router({
+    collectivites : this.trpc.authedProcedure
+      .input(filtersRequestSchema)
+      .query(async ({input}) => {
+        return await this.service.collectivites(input);
+      }),
```

Comment:
C'est un détail, mais il n'est pas nécessaire de mettre async/await ici puisque c'est la promise qui est renvoyé directement.
On peut écrire juste `.query({input}) => { return this.service.collectivites(input); }` (voir même `.query({input}) => this.service.collectivites(input)`) et ça fait la même chose.